### ISSUE TITLE
实验复现

### ISSUE THREAD

**Issue by 611699** [demonph10/PGL#1]
作者您好，您的代码我很难达到你论文提供的程度，或许是我的超参数设置不对，你能提供一下能大致复现实验指标的超参数吗，针对baby数据集的就行

**Comment by demonph10** [comment]
您好，PGL w/global 在baby上的实验记录如下，可以参考其中的超参数设置。如还有疑问，欢迎进一步探讨。
Wed 03 Jul 2024 14:45:29 INFO 
gpu_id=0
use_gpu=True
seed=[999]
data_path=../data/
inter_splitting_label=x_label
filter_out_cod_start_users=True
is_multimodal_model=True
checkpoint_dir=saved
save_recommended_topk=True
recommend_topk=recommend_topk/
embedding_size=64
weight_decay=0.0
req_training=True
epochs=1000
stopping_step=20
train_batch_size=2048
learner=adam
learning_rate=0.001
learning_rate_scheduler=[0.96, 50]
eval_step=1
training_neg_sample_num=1
use_neg_sampling=True
use_full_sampling=False
NEG_PREFIX=neg__
USER_ID_FIELD=userID
ITEM_ID_FIELD=itemID
TIME_FIELD=timestamp
field_separator=	
metrics=['Recall', 'NDCG']
topk=[5, 10, 20, 50]
valid_metric=Recall@20
eval_batch_size=2048
use_raw_features=False
max_txt_len=32
max_img_size=256
vocab_size=30522
type_vocab_size=2
hidden_size=4
pad_token_id=0
max_position_embeddings=512
layer_norm_eps=1e-12
hidden_dropout_prob=0.1
end2end=False
hyper_parameters=['dropout', 'reg_weight', 'seed']
inter_file_name=baby.inter
vision_feature_file=image_feat.npy
text_feature_file=text_feat.npy
user_graph_dict_file=user_graph_dict.npy
feat_embed_dim=64
weight_size=[64, 64]
lambda_coeff=0.9
reg_weight=[0.05]
n_mm_layers=1
n_ui_layers=2
knn_k=10
mm_image_weight=0.1
dropout=[0.2]
model=PGL
dataset=baby
valid_metric_bigger=True
device=cuda

....

█████████████ BEST ████████████████
Wed 03 Jul 2024 15:00:40 INFO 	Parameters: ['dropout', 'reg_weight', 'seed']=(0.2, 0.05, 999),
Valid: recall@5: 0.0401    recall@10: 0.0625    recall@20: 0.0999    recall@50: 0.1663    ndcg@5: 0.0263    ndcg@10: 0.0336    ndcg@20: 0.0431    ndcg@50: 0.0564    ,
Test: recall@5: 0.0412    recall@10: 0.0663    recall@20: 0.1040    recall@50: 0.1695    ndcg@5: 0.0269    ndcg@10: 0.0351    ndcg@20: 0.0448    ndcg@50: 0.0581

**Comment by 611699** [comment]
非常感谢您的回复，我使用了您提供的超参数，效果确实好了不少，但是距离你的实验结果依然有不少距离，并且我发现你的超参数里面并没有mode=['local']，这是什么原因呢

**Comment by 611699** [comment]
另外，能否问一下global模式下的相关超参数吗，我觉得这个想法很有意思，想探讨一下

**Comment by demonph10** [comment]
> 您好，PGL w/global 在baby上的实验记录如下，可以参考其中的超参数设置。如还有疑问，欢迎进一步探讨。 Wed 03 Jul 2024 14:45:29 INFO gpu_id=0 use_gpu=True seed=[999] data_path=../data/ inter_splitting_label=x_label filter_out_cod_start_users=True is_multimodal_model=True checkpoint_dir=saved save_recommended_topk=True recommend_topk=recommend_topk/ embedding_size=64 weight_decay=0.0 req_training=True epochs=1000 stopping_step=20 train_batch_size=2048 learner=adam learning_rate=0.001 learning_rate_scheduler=[0.96, 50] eval_step=1 training_neg_sample_num=1 use_neg_sampling=True use_full_sampling=False NEG_PREFIX=neg__ USER_ID_FIELD=userID ITEM_ID_FIELD=itemID TIME_FIELD=timestamp field_separator= metrics=['Recall', 'NDCG'] topk=[5, 10, 20, 50] valid_metric=Recall@20 eval_batch_size=2048 use_raw_features=False max_txt_len=32 max_img_size=256 vocab_size=30522 type_vocab_size=2 hidden_size=4 pad_token_id=0 max_position_embeddings=512 layer_norm_eps=1e-12 hidden_dropout_prob=0.1 end2end=False hyper_parameters=['dropout', 'reg_weight', 'seed'] inter_file_name=baby.inter vision_feature_file=image_feat.npy text_feature_file=text_feat.npy user_graph_dict_file=user_graph_dict.npy feat_embed_dim=64 weight_size=[64, 64] lambda_coeff=0.9 reg_weight=[0.05] n_mm_layers=1 n_ui_layers=2 knn_k=10 mm_image_weight=0.1 dropout=[0.2] model=PGL dataset=baby valid_metric_bigger=True device=cuda
> 
> ....
> 
> █████████████ BEST ████████████████ Wed 03 Jul 2024 15:00:40 INFO Parameters: ['dropout', 'reg_weight', 'seed']=(0.2, 0.05, 999), Valid: recall@5: 0.0401 recall@10: 0.0625 recall@20: 0.0999 recall@50: 0.1663 ndcg@5: 0.0263 ndcg@10: 0.0336 ndcg@20: 0.0431 ndcg@50: 0.0564 , Test: recall@5: 0.0412 recall@10: 0.0663 recall@20: 0.1040 recall@50: 0.1695 ndcg@5: 0.0269 ndcg@10: 0.0351 ndcg@20: 0.0448 ndcg@50: 0.0581

这个就是global的版本，超参数中没有mode参数是因为这是去年（Wed 03 Jul 2024 15:00:40）进行实验时的实验记录，还未合并为最终发布的代码。如果要在我们发布的代码上复现global版本，应该将mode设置为global。
